### PR TITLE
LibLine: ^X^E - edit in external editor

### PR DESCRIPTION
**Pull request by alimpfard**
Adds support for ^X^E, not sure how we've been living without this so far!

Also fixes a dumb case where LibLine would miss a WINCH signal while another program is in the foreground.
It can be reproduced like so (prior to the changes in this PR)
- `$ te`
- resize the terminal while `te` is running
- close `te`
- now the shell doesn't know that the terminal size has changed

### CODE REVIEW

File: Userland/Libraries/LibLine/Editor.cpp [hunk 1]
```diff
@@ -168,6 +174,9 @@ void Editor::set_default_keybinds()
     register_key_input_callback(ctrl('T'), EDITOR_INTERNAL_FUNCTION(transpose_characters));
     register_key_input_callback('\n', EDITOR_INTERNAL_FUNCTION(finish));
 
+    // ^X^E: Edit in external editor
+    register_key_input_callback(Vector<Key> { ctrl('X'), ctrl('E') }, EDITOR_INTERNAL_FUNCTION(edit_in_external_editor));
```
Comment (linusg) [inline Userland/Libraries/LibLine/Editor.cpp:178]: I've never looked at LibLine in detail, that's a nice API you got there! :D

File: Userland/Libraries/LibLine/InternalFunctions.cpp [hunk 2]
```diff
@@ -522,4 +526,91 @@ void Editor::uppercase_word()
...
+        perror("mktemp");
+        return;
+    }
+
+    {
+        auto* fp = fdopen(fd, "rw");
+        if (!fp) {
+            perror("fdopen");
+            return;
+        }
```
Comment (awesomekling) [inline Userland/Libraries/LibLine/InternalFunctions.cpp:548]: This gunk makes me feel like we should have something in LibCore for creating a temp file.

Comment (IdanHo) [reply Userland/Libraries/LibLine/InternalFunctions.cpp:548]: we have similar (but slightly different) gunk in https://github.com/SerenityOS/serenity/blob/master/Userland/Utilities/mktemp.cpp

Comment (alimpfard) [reply Userland/Libraries/LibLine/InternalFunctions.cpp:548]: Definitely should, +1.

**Pull request merged**